### PR TITLE
Fix self-dependency versions

### PR DESCRIPTION
**Pull request by andrewbranch**
Ensures packages don’t depend on old versions of each other.

### CODE REVIEW

File: packages/dtslint/src/index.ts [hunk 1]
```diff
@@ -20,6 +20,8 @@ async function main(): Promise<void> {
   let lookingForTsLocal = false;
   let tsLocal: string | undefined;
 
+  console.log(`dtslint@${require("../package.json").version}`);
```
Comment (sandersn) [inline packages/dtslint/src/index.ts:23]: is this helpful long-term?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,6 +17,19 @@ jobs:
     - uses: actions/setup-node@v2
       with:
         node-version: ${{ matrix.node }}
+    - name: check dependencies
+      run: |
+        if cat yarn.lock | grep "@definitelytyped/"; then
```
Comment (sandersn) [inline .github/workflows/ci.yml:22]: isn't this equivalent?
```suggestion
        if grep "@definitelytyped/" yarn.lock; then
```

I don't understand the scope of this command -- isn't this the root yarn.lock? How does this check the package yarn.locks? Or is yarn.lock a convenient single point where all incorrect dependencies show up?

**Review by sandersn — CHANGES_REQUESTED** [review]
Couple of questions

File: .github/workflows/ci.yml (hunk 2 above)
Comment (andrewbranch) [reply .github/workflows/ci.yml:22]: For workspaces, there's a single yarn.lock.

And yes, I'm bad at bash.

File: packages/dtslint/src/index.ts (hunk 1 above)
Comment (andrewbranch) [reply packages/dtslint/src/index.ts:23]: I think so. I added it when the version mismatch was just a hunch, not yet confirmed, and not yet mitigated with the CI check. But I think it’s useful to be able to look at older logs and tell what was running.

**Review by sandersn — APPROVED** [review]

**Pull request merged**